vdom/apply-properties.js

```diff
@@ -1,100 +1,116 @@
-var isObject = require("is-object")
-var isHook = require("../vnode/is-vhook.js")
+var isObject = require('is-object');
+var isHook = require('../vnode/is-vhook');
+var isSoftSetHook = require('./is-soft-set-hook');
+var undefinedValue = require('./undefined-value');
 
-module.exports = applyProperties
+module.exports = applyProperties;
 
 function applyProperties(node, props, previous) {
-    for (var propName in props) {
-        var propValue = props[propName]
-
-        if (propValue === undefined) {
-            removeProperty(node, propName, propValue, previous);
-        } else if (isHook(propValue)) {
-            removeProperty(node, propName, propValue, previous)
-            if (propValue.hook) {
-                propValue.hook(node,
-                    propName,
-                    previous ? previous[propName] : undefined)
-            }
-        } else {
-            if (isObject(propValue)) {
-                patchObject(node, props, previous, propName, propValue);
-            } else {
-                try {
-                    // this is a workaround in case of invalid value or readonly propName
-                    node[propName] = propValue;
-                } catch(err) {}
-            }
-        }
+  for (var propName in props) {
+    var propValue = props[propName];
+
+    if (undefinedValue.isUndefined(propValue)) {
+      removeProperty(node, propName, propValue, previous);
+
+    } else if (isHook(propValue)) {
+      removeProperty(node, propName, propValue, previous);
+
+      if (propValue.hook) {
+        propValue.hook(node,
+          propName,
+          previous ? previous[propName] : undefined);
+      }
+
+    } else if (isSoftSetHook(propValue)) {
+      removeProperty(node, propName, propValue, previous);
+      node[propName] = propValue.value;
```

Comment:
use setProperty method here and on line 55 and 57 as well

Reply:
ok